lore/io/__init__.py

```diff
@@ -86,7 +86,7 @@ def download(remote_url, local_path=None, cache=True, extract=False):
             raise NotImplementedError("Cannot download from s3 without config/aws.cfg")
         protocol = 's3'
         remote_url = prefix_remote_root(remote_url)
-        _bucket = None
+        _bucket = bucket
```

Comment:
This will still fail if we try to download a file when the bucket is not set in aws.cfg. Would it make sense to error if the bucket is not set?

Reply:
Basically this flow Line 67 -> 85 -> 89

Reply:
You're  right @sarchak I think this case was present in the original code though. I will fix.